src/Disks/createVolume.cpp

```diff
@@ -15,18 +17,18 @@ namespace ErrorCodes
     extern const int INVALID_RAID_TYPE;
 }
 
-VolumePtr createVolumeFromReservation(const ReservationPtr & reservation, VolumePtr other_volume)
+VolumePtr createVolumeFromReservation(const ReservationPtr & reservation, VolumePtr original_volume)
 {
-    if (other_volume->getType() == VolumeType::JBOD || other_volume->getType() == VolumeType::SINGLE_DISK)
+    if (original_volume->getType() == VolumeType::JBOD || original_volume->getType() == VolumeType::SINGLE_DISK)
     {
         /// Since reservation on JBOD chooses one of disks and makes reservation there, volume
         /// for such type of reservation will be with one disk.
-        return std::make_shared<SingleDiskVolume>(other_volume->getName(), reservation->getDisk(), other_volume->max_data_part_size);
+        return std::make_shared<SingleDiskVolume>(original_volume->getName(), reservation->getDisk());
```

Comment:
So instead of avoiding this completely you decided to provide an interface that will throw LOGICAL_ERROR? I though that you were talking about removing this completely (provide reservation all the knowledge for the callers of this helper)